plugins/inputs/nats_consumer/nats_consumer.go

```diff
@@ -142,6 +145,31 @@ func (n *natsConsumer) Start(acc telegraf.Accumulator) error {
 
 			n.subs = append(n.subs, sub)
 		}
+
+		if len(n.JsSubjects) > 0 {
+			var connErr error
+			n.jsConn, connErr = n.conn.JetStream(nats.PublishAsyncMaxPending(256))
+			if connErr != nil {
+				return connErr
+			}
+
+			for _, jsSub := range n.JsSubjects {
+				sub, err := n.jsConn.QueueSubscribe(jsSub, n.QueueGroup, func(m *nats.Msg) {
+					n.in <- m
+				})
+				if err != nil {
+					return err
+				}
+
+				// set the subscription pending limits
+				err = sub.SetPendingLimits(n.PendingMessageLimit, n.PendingBytesLimit)
+				if err != nil {
+					return err
+				}
+
+				n.jsSubs = append(n.jsSubs, sub)
+			}
+		}
```

Comment:
Shouldn't this be guarded with a `if n.jsConn == nil || n.jsConn.IsClosed() {` statement similar to `n.conn`?

Reply:
n.jsConn.IsClosed is not availabe but I adden if n.jsConn == nil